### ISSUE TITLE
BUG - Invalid empty identifier

### ISSUE THREAD

**Issue by kevinohara80** [fivetran/dbt_salesforce_formula_utils#22]
**Are you a current Fivetran customer?** 
Yes, Kevin O'Hara @ Signal Advisors

**Describe the bug**
Getting the following error when creating a model with `sfdc_formula_view`

```
Completed with 1 error and 0 warnings:

Database Error in model lead_view (models/sfdc_production/lead_view.sql)
  Syntax error: Invalid empty identifier at [257:156]
  compiled SQL at target/run/fivetran_dbt/models/sfdc_production/lead_view.sql

Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

The referenced line in the compile SQL shows the issue with the missing identifier:

```
 ... from `our-bigquery-project`.`sfdc_production`.``lead left join ...
```

I'm assuming that the table name `lead` should render inside of the backticks?

**Steps to reproduce**
1. Run `dbt run` against your bigquery target

**Expected behavior**
Should render the table name inside of the backticks

**Project variables configuration**

```yml
# Name your project! Project names should contain only lowercase characters
# and underscores. A good package name should reflect your organization's
# name or the intended use of these models
name: "fivetran_dbt"
version: "1.0.0"
config-version: 2

# This setting configures which "profile" dbt uses for this project.
profile: bigquery

# These configurations specify where dbt should look for different types of files.
# The `source-paths` config, for example, states that models in this project can be
# found in the "models/" directory. You probably won't need to change these!
source-paths: ["models"]
analysis-paths: ["analysis"]
test-paths: ["tests"]
data-paths: ["data"]
macro-paths: ["macros"]
snapshot-paths: ["snapshots"]

target-path: "target" # directory which will store compiled SQL files
clean-targets: # directories to be removed by `dbt clean`
  - "target"
  - "dbt_modules"

# Configuring models
# Full documentation: https://docs.getdbt.com/docs/configuring-models

# In this example config, we tell dbt to build all models in the example/ directory
# as tables. These settings can be overridden in the individual model files
# using the `{{ config(...) }}` macro.
models:
  fivetran_dbt:
    sfdc_production:
      +enabled: true
      +materialized: view
```

**Package Version**

```yml
packages:
  - package: fivetran/salesforce_formula_utils
    version: 0.4.0

  - package: dbt-labs/dbt_utils
    version: 0.7.0
```

**Warehouse**

 
- [x] BigQuery
- [ ] Redshift
- [ ] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Additional context**

Model file (lead_view.sql):

```
{{ salesforce_formula_utils.sfdc_formula_view(source_table='lead') }}
```

Source file(sfdc_production.yml):

```yml
version: 2

sources:
  - name: salesforce #This should always be salesforce
    schema: sfdc_production #Modify this to be where your Salesforce data resides
    database: #Add the database where your Salesforce data resides. Eg. 'my_salesforce_database'. By default the target.database is used.

    #This source yml contains the following Salesforce tables: fivetran_formula, opportunity, account, user, user_role.
    #If you would like to include additional source tables you can copy add them in a similar fashion, then paste within your
    #root dbt project. Similarly, if you do not have these tables you can remove them.

    tables:
      - name: fivetran_formula
        description: A table containing a mapping of sfdc formulas with a reference to corresponding table objects for the formulas to be applied.
        columns:
          - name: field
            description: Name of the formula
          - name: object
            description: Reference to the table the formula is to be applied
          - name: formula
            description: Salesforce defined formula
          - name: sql
            description: Standard sql interpretation of the Salesforce defined formula, generated by Fivetran
          - name: _fivetran_synced
            description: Timestamp of when the record was synced by Fivetran

      - name: account
        description: Represents an individual account, which is an organization or person involved with your business (such as customers, competitors, and partners).
        columns:
          - name: id
            description: The unique, system-generated ID assigned during creation

      - name: opportunity
        description: Represents an opportunity, which is a sale or pending deal.
        columns:
          - name: id
            description: The unique, system-generated ID assigned during creation.

      - name: case
        description: Represents a case in your organization.
        columns:
          - name: id
            description: The unique, system-generated ID assigned during creation.

      - name: contact
        description: Represents a contact in your organization.
        columns:
          - name: id
            description: The unique, system-generated ID assigned during creation.

      - name: lead
        description: Represents a lead in your organization.
        columns:
          - name: id
            description: The unique, system-generated ID assigned during creation.

      - name: user
        description: Represents a user in your organization.
        columns:
          - name: id
            description: The unique, system-generated ID assigned during creation.

      - name: user_role
        description: Represents a user role in your organization.
        columns:
          - name: id
            description: The unique, system-generated ID assigned during creation

```

**Screenshots**

**Please indicate the level of urgency** 
Unable to make progress on our project as a whole

**Are you interested in contributing to this package?** 

- [ ] Yes, I can do this and open a PR for your review.
- [x] Possibly, but I'm not quite sure how to do this. I'd be happy to do a live coding session with someone to get this fixed. 
- [ ] No, I'd prefer if someone else fixed this.  I don't have the time and/or don't know what the root cause of the problem is.

**Comment by fivetran-jamie** [comment]
hey @kevinohara80 sorry you're seeing this issue! we'll look into this early next week and get back to you here

**Comment by kevinohara80** [comment]
@fivetran-jamie Thank you! We're looking forward to getting this working!

**Comment by fivetran-joemarkiewicz** [comment]
@kevinohara80 As this package was originally developed using Snowflake, we did not anticipate the backtick adjustment for BigQuery users. I have since been able to create a working branch and a draft PR that should resolve this issue for you. You can give the below a try in your `packages.yml` and let us know if this fixes the issue.
```yml
packages:
    - git: https://github.com/fivetran/dbt_salesforce_formula_utils.git
      revision: fix/bigquery-backtick
      warn-unpinned: false
```

Additionally, if you experience an alias error within your `view_sql` please know that I have opened an internal ticket with our product team to help address this issue. If that is the case, you can exclude failing formulas using the `sfdc_exclude_formulas` variable.

Let me know if this helps!

**Comment by kevinohara80** [comment]
@fivetran-joemarkiewicz Thanks for looking into this. That branch seems to resolve the current issue, but I'm running into a different issue now.

Given

```
{{ salesforce_formula_utils.sfdc_formula_view(source_table='lead', fields_to_include=('weighted_pipeline_c')) }}
```

I get the following error:

```
Running with dbt=0.20.0
Found 1 model, 0 tests, 0 snapshots, 0 analyses, 430 macros, 0 operations, 0 seed files, 16 sources, 0 exposures

10:37:47 | Concurrency: 1 threads (target='dev')
10:37:47 | 
10:37:47 | 1 of 1 START view model dbt_dev.lead_view............................ [RUN]
10:37:49 | 1 of 1 ERROR creating view model dbt_dev.lead_view................... [ERROR in 2.34s]
10:37:49 | 
10:37:49 | Finished running 1 view model in 4.41s.

Completed with 1 error and 0 warnings:

Database Error in model lead_view (models/sfdc_production/lead_view.sql)
  Syntax error: Expected "(" or keyword UNNEST but got identifier "weighted_pipeline_c" at [13:30]

Done. PASS=0 WARN=0 ERROR=1 SKIP=0 TOTAL=1
```

Any idea what that error could mean?

**Comment by fivetran-joemarkiewicz** [comment]
Ah yes I came across this with BQ as well. Could you try the below macro instead?

```sql
{{ salesforce_formula_utils.sfdc_formula_view(source_table='lead', fields_to_include="('weighted_pipeline_c')") }}
```
I found closing the `fields_to_include` argument in quotes was needed. I am still looking into why this is required for BigQuery, but I did find it resolved on my end when this was done. Let me know if this works!

**Comment by kevinohara80** [comment]
So, it looks like that issue presents itself when there is only one item in `fields_to_include`. If I add a second formula, the view is created successfully.

**Comment by fivetran-joemarkiewicz** [comment]
Oh interesting 🤔 

I am glad the view was created succesfull though!! I will move forward with integrating the working branch into the next release. I will let you know here once we cut a new patch release of the package.

**Comment by fivetran-joemarkiewicz** [comment]
Hi @kevinohara80 I just merged and cut a new release `v0.4.1` for this package which addresses the backtick issue you were seeing. This release should be live on the dbt hub at the top of the hour. 

If you have any other issues or a feature request, feel free to open another issue. Thank you again for raising this with our team!
